src/main/java/org/elasticsearch/index/gateway/IndexShardGateway.java

```diff
@@ -105,8 +95,7 @@ public IndexShardGateway(ShardId shardId, @IndexSettings Settings indexSettings,
     public void recover(boolean indexShouldExists, RecoveryState recoveryState) throws IndexShardGatewayRecoveryException {
         indexShard.prepareForIndexRecovery();
         long version = -1;
-        long translogId = -1;
-        final Set<String> typesToUpdate = Sets.newHashSet();
+        Set<String> typesToUpdate = Sets.newHashSet();
```

Comment:
I think this can be set to final and not intialized (`final Set<String> typesToUpdate;`) . It's only set once and it has to be set now.